### PR TITLE
I2c/testing

### PR DESCRIPTION
**Pull request by vireshk**
Lets improve coverage score a bit more :)

### CODE REVIEW

**Review by mathieupoirier — COMMENTED** [review]
The Option() on the VringRwLock is not mandatory and this patch is essentially a code cleanup, something the changelog should clearly state.  I kept looking for a higher purpose to this work but in the end there isn't.

**Review by mathieupoirier — APPROVED** [review]
Ideally this patch would have been split in 3 different patches, a patch for each of the clearly defined section.  I agree that each patch would have been small but that is fine.  It would have sped up the the review process by making it clear each patch is unrelated to the other.  Please consider for next time.

File: i2c/src/vhu_i2c.rs [hunk 1]
```diff
@@ -114,7 +114,7 @@ impl<D: I2cDevice> VhostUserI2cBackend<D> {
     fn process_requests(
         &self,
         requests: Vec<I2cDescriptorChain>,
-        vring: Option<&VringRwLock>,
+        vring: &VringRwLock,
```
Comment (mathieupoirier) [inline i2c/src/vhu_i2c.rs:117]: The Option() on the VringRwLock is not mandatory and this patch is essentially a code cleanup, something the changelog should clearly state. I kept looking for a higher purpose to this work but in the end there isn't.

File: coverage_config_x86_64.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "coverage_score": 73.9,
+  "coverage_score": 86.5,
```
Comment (mathieupoirier) [inline coverage_config_x86_64.json:2]: Ideally this patch would have been split in 3 different patches, a patch for each of the clearly defined section. I agree that each patch would have been small but that is fine. It would have sped up the the review process by making it clear each patch is unrelated to the other. Please consider for next time.

**Review by mathieupoirier — APPROVED** [review]

File: i2c/src/vhu_i2c.rs (hunk 1 above)
Comment (vireshk) [reply i2c/src/vhu_i2c.rs:117]: Hmm, it was actually essential until this patch came in as I didn't figure out earlier that we can actually send a empty vring from tests to make it run and so the tests were rather passing None earlier.. And this is what the changelog says now:

    For tests we can pass a non related vring to just make it run, hence no
    need of making it optional. Always pass the vring.

I hoped it was alright.

File: coverage_config_x86_64.json (hunk 2 above)
Comment (vireshk) [reply coverage_config_x86_64.json:2]: You mean separate patches for each of the three .rs files? Yeah, that could have been done well to make the patches smaller. Will try to do that in future.

**Comment by vireshk** [comment]
Thanks for your reviews @mathieupoirier

**Review by stsquad — APPROVED** [review]
LGTM! :thumbsup:

File: i2c/src/vhu_i2c.rs (hunk 1 above)
Comment (mathieupoirier) [reply i2c/src/vhu_i2c.rs:117]: I have read the changelog several times again today and I still find it difficult to relate it to the code presented by the patch.  But that is my own interpretation, i.e what is clear for you might not be for me and vise versa.  In the end it is just terminology and that is why I didn't think is was necessary to delay approval of the patch.

File: coverage_config_x86_64.json (hunk 2 above)
Comment (mathieupoirier) [reply coverage_config_x86_64.json:2]: Right, the enhancements in each file are independent from each other and could have been split in three patches, with each patch updating its own coverage score.  I found myself looking for a relation between the hunks but there wasn't any.  But again not important enough to delay this patchset or spent any more time talking about it.
